### DISCUSSION TITLE
pre-commit failed

### DISCUSSION THREAD

**Discussion by willeke1234** [GrandOrgue/grandorgue#1993] [Developers]
Hi,

I'm trying to do a pull request but pre-commit failed. The copyright year is 2023 and there is a formatting issue. What did I miss and how do I fix this? Git is still a mystery to me.

**Comment by rousseldenis** [comment]
> Hi,
> 
> I'm trying to do a pull request but pre-commit failed. The copyright year is 2023 and there is a formatting issue. What did I miss and how do I fix this? Git is still a mystery to me.

You should install pre-commit locally.

If there are formatting issues, it will say to you at commit (but forbid to do it).

Usually it changes automatically the code. You need to git add one more time, then git commit. And voila

**Comment by willeke1234** [comment]
update-copyright passed but clang-format failed.

```
clang-format.............................................................Failed
- hook id: clang-format
- exit code: 1

Traceback (most recent call last):
  File "/Users/gebruiker/.cache/pre-commit/repoleyp1c2j/py_env-python3.12/bin/clang-format", line 8, in <module>
    sys.exit(clang_format())
             ^^^^^^^^^^^^^^
  File "/Users/gebruiker/.cache/pre-commit/repoleyp1c2j/py_env-python3.12/lib/python3.12/site-packages/clang_format/__init__.py", line 11, in clang_format
    return _run("clang-format")
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/gebruiker/.cache/pre-commit/repoleyp1c2j/py_env-python3.12/lib/python3.12/site-packages/clang_format/__init__.py", line 7, in _run
    return os.execv(executable, [name] + sys.argv[1:])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 86] Bad CPU type in executable
```
Do I need to install anything else? A working clang-format is already installed.

**Reply by oleg68** [reply]
`Bad CPU type in executable` means that you have installed `clang-format`, but for another CPU: arm64 instead of amd64 or vice versa. 

1. Could you run `clang-format` manually from the command line?
2. pre-commit automatically tries to download clang-format under the `.cache` directory. Possible it downloaded it for wrong CPU. Could you try to remove the `.cache` directory and to execute `pre-commit run -a`?
3. If it still failed then pre-commit has a bug: it downloads executables for wrong cpus. Could you a workaround: to install clang-format of right cpu manually?

**Comment by willeke1234** [comment]
I downloaded clang-format and git-clang-format again and replaced them in the .cache directory again and now it works. I thought I had already tried that but I probably did something different this time.
Thanks for your help.
